packages/apps/api/src/handlers/user.ts

```diff
@@ -118,32 +130,68 @@ export async function updateUser(sierraClient: SierraClient, auth0Client: Auth0C
   }
 
   const email: string = request.body.email;
-  if (!isNonBlank(email)) {
-    response.status(400).json(toMessage('All fields must be provided and non-blank'));
+  if (email && !isNonBlank(email)) {
+    response.status(400).json(toMessage('email must be a non-blank string'));
+    return;
+  }
+
+  const firstName: string = request.body.firstName;
+  if (firstName && !isNonBlank(firstName)) {
+    response.status(400).json(toMessage('firstName must be a non-blank string'));
+    return;
+  }
+
+  const lastName: string = request.body.lastName;
+  if (lastName && !isNonBlank(lastName)) {
+    response.status(400).json(toMessage('lastName must be a non-blank string'));
+    return;
+  }
+
+  if ((firstName || lastName) && !userIsAdmin(request)) {
+    response.status(403).json(toMessage(`Only an administrator can change ${privilegedUserFields.join(', ')}`));
     return;
   }
 
   const auth0Get: APIResponse<Auth0Profile> = await auth0Client.getUserByEmail(email);
   if (auth0Get.status != ResponseStatus.NotFound) {
-    if (auth0Get.status === ResponseStatus.Success) {
+    if (auth0Get.status === ResponseStatus.Success && auth0Get.result.userId !== userId) {
       response.status(409).json(toMessage('Auth0 user with email [' + email + '] already exists'));
-    } else {
+      return;
+    } else if (auth0Get.status == ResponseStatus.UnknownError) {
```

Comment:
I added a `userId === profile.userId` check to the first condition, so that else will also be true for `status === ResponseStatus.Success`. I don't think this can return anything other than NotFound, Success, or an UnknownError.